android/tests_backend/app.py

```diff
@@ -42,6 +42,11 @@ def logs_path(self):
     def assert_app_icon(self, icon):
         pytest.xfail("Android apps don't have app icons at runtime")
 
+    def assert_dialog_in_focus(self, dialog):
+        assert (
+            dialog._impl.native_alert_dialog.isShowing() is True
```

Comment:
Related to above - why isn't this just `dialog._impl.native.isShowing()`?

Reply:
Same reason as above: https://github.com/beeware/toga/pull/2926/files#r1834096789

`AlertDialog` has the `isShowing()` method: https://developer.android.com/reference/android/app/Dialog#isShowing()
which is used in the probe.